online/src/main/scala/ai/chronon/online/Api.scala

```diff
@@ -265,4 +268,35 @@ abstract class Api(userConf: Map[String, String]) extends Serializable {
     new Consumer[LoggableResponse] {
       override def accept(t: LoggableResponse): Unit = logResponse(t)
     }
+
+  /**
+    * Helper function to get join configuration with error handling.
+    *
+    * @param joinName Name of the join to load configuration for
+    * @return JoinOps for the specified join
+    * @throws RuntimeException if join configuration fails to load
+    */
+  protected def getJoinConfiguration(joinName: String): JoinOps = {
+    val joinConfTry = fetcher.getJoinConf(joinName)
+    joinConfTry match {
+      case Success(joinOps) => joinOps
+      case Failure(exception) =>
+        logger.error(s"Failed to load join configuration for '$joinName': ${exception.getMessage}", exception)
+        throw new RuntimeException(s"Failed to load join configuration for '$joinName'", exception)
+    }
+  }
+
+  /**
+    * Register external sources from join configurations.
+    * This function reads configurations for each join, extracts FactoryConfig information
+    * from external sources, and stores this information in the ExternalSourceRegistry.
+    *
+    * @param joins Sequence of join names to process
+    */
+  def registerExternalSources(joins: Seq[String]): Unit = {
```

Comment:
even though ES is only available in Join, but let's still add a join to the method name. 
```suggestion
  def registerJoinExternalSources(joins: Seq[String]): Unit = {
```